controllers/secretgenerator/secret.go

```diff
@@ -90,7 +90,7 @@ func generateSecretValue(secret *Secret) error {
 	switch secret.Type {
 	case "random":
 		randomValue := make([]byte, secret.Complexity)
-		for i := 0; i < secret.Complexity; i++ {
+		for i := range secret.Complexity {
```

Comment:
This code adjustments are not mandatory for the version upgrade, right?

Reply:
I guess not 100% mandatory, but without them the linter check would fail, so we'd need to disable some lint checks. Since there aren't that many, and they're simple, I figured that it would be preferable to just fix them and include that in the PR.

On the subject of what's mandatory and not: hypothetically we could choose not to bump the version of Go in go.mod (which would mean not having access to these new language features), and just bump it in the Dockerfiles and automation (done). I figured why not upgrade that too though, because it seems straightforward.